src/Client.js

```diff
@@ -14,22 +14,51 @@ class Client {
     this.options = {
       cache: options.cache || false,
       cacheTime: options.cacheTime || 60,
-      cacheSize: options.cacheSize || -1
+      cacheSize: options.cacheSize || -1,
+      rateLimit: options.rateLimit || 'AUTO'
     };
     this.key = key;
+    this.requests = 0;
+    this.lastRequestAt = 9999999999999; // Set to a large number so 1st request doesn't get rate limited already
     this._validateOptions();
+    if (this.options.rateLimit !== 'NONE') {
+      this.getKeyInfo().then(info => {
+        this.requests = info.requestsInPastMin;
+        setTimeout(this._rateLimitMonitor, 1000 * info.resetsAfter);
+      });
+    }
+  }
+
+  _rateLimitMonitor () {
+    this.requests = 0;
+    // eslint-disable-next-line no-return-assign
+    setInterval(() => this.requests = 0, 1000 * 60);
+  }
+
+  _rateLimitManager () {
+    // eslint-disable-next-line no-useless-return
+    if (this.options.rateLimit === 'AUTO' && this.requests <= 60) return;
+    if (new Date().getTime() - this.lastRequestAt >= 500) return;
+    // Wait before send, because user is on HARD RateLimit mode or AUTO, but passed 60 requests/min
+    // With rate limit set to HARD, you will never be able to pass the Ratelimit set by hypixel API if this is the only script you are using the API key with.
+    // eslint-disable-next-line promise/param-names
+    return new Promise(r => setTimeout(r, 500));
```

Comment:
Correct me if I'm wrong but I don't think this will work as intended, because three (or more) requests within the same 500ms will result in two (or more - 1) of them just being delayed for the same time, thus allowing more than 120 reqs/min even in 'HARD' mode

Reply:
right... i see, will work on patcj